### PR TITLE
Design doc

### PR DESCRIPTION
**Pull request by mccallm1**
Only major changes to design.tex and removed extra files. Please look over changes to 3.1, 3.1.1, and 3.3.1.

### CODE REVIEW

File: docs/design_document/design.tex [hunk 1]
```diff
@@ -69,10 +69,14 @@ \section{Definitions}
...
+Ubuntu is the operating system used on our project to control both the intelligent agent and all computers that communicate with it. The intelligent agent's on-board OS functions like any other computer, but runs a specialized program called ROS (Robot Operating System) to control all "robotic" functions. The computer used to communicate with the intelligent agent must also run Ubuntu and ROS, as the software used alongside ROS must operate through it. Programs such as Rviz, Gmapping, and Mapsaver are all used as auxilary programs to aid us in gathering our test data.
```
Comment (julianweisbord) [inline docs/design_document/design.tex]: I think this paragraph is kind of awkward. What if we changed it to something like:

Ubuntu is the operating system (OS) used on our project to control both the intelligent agent and all computers that communicate with it. The intelligent agent's on-board OS functions like any other computer, but runs ROS (Robot Operating System) which is an Open Source set of libraries that provide a series of high-level functions to help user's write software for mobile robots. The computer used to communicate with the intelligent robot must also run Ubuntu and ROS to create an effective channel for robot communication and data transfer. Programs such as Rviz, Gmapping, and Mapsaver are all used as auxilary programs to aid us in gathering our test data.

**Comment by julianweisbord** [comment]
Remember to add reviewers next time.

File: docs/design_document/design.tex [hunk 2]
```diff
@@ -69,10 +69,14 @@ \section{Definitions}
...
+Ubuntu is the operating system used on our project to control both the intelligent agent and all computers that communicate with it. The intelligent agent's on-board OS functions like any other computer, but runs a specialized program called ROS (Robot Operating System) to control all "robotic" functions. The computer used to communicate with the intelligent agent must also run Ubuntu and ROS, as the software used alongside ROS must operate through it. Programs such as Rviz, Gmapping, and Mapsaver are all used as auxilary programs to aid us in gathering our test data. 
+
+While ROS is not a true operating system in the same sense Linux is, the program runs constantly in the background of the intelligent agent managing everything from sensors, to motors, to inter-process communication. Compontents of the intelligent agent are registered as devices under ROS and controlled similarly to typical device drivers.
```
Comment (julianweisbord) [inline docs/design_document/design.tex]: I don't think there is a big enough relation between ROS and Linux to say this, ROS is a framewok of tools and software packages for controlling robots while Linux is a kernel which connects GUI applications to the CPU, RAM, I/O, etc.

File: docs/design_document/design.tex [hunk 3]
```diff
@@ -69,10 +69,14 @@ \section{Definitions}
...
+
+While ROS is not a true operating system in the same sense Linux is, the program runs constantly in the background of the intelligent agent managing everything from sensors, to motors, to inter-process communication. Compontents of the intelligent agent are registered as devices under ROS and controlled similarly to typical device drivers. 
+
+ROS is also the most common operating system for robotic systems. It is open source and has BSD licensing which are two big reasons we chose it as part of our design. Being open source is one of the main reasons that RuOS has evolved so quickly and become so prominent in the field of robotics research and development. Two key concepts exist that stand out more than the others and that is the package and the stack. The package is a directory that has external libraries, API data, configuration files, nodes and an XML configuration file. The stack is a collection of packages that offer functionalities like navigation, positioning, mapping, and others \cite{ROSintro}.
```
Comment (julianweisbord) [inline docs/design_document/design.tex]: RuOS?

Comment (julianweisbord) [reply docs/design_document/design.tex]: Consider revising: "Two key concepts exist that stand out more than the others and that is the package and the stack."

Also maybe change: "The stack is a collection of packages that offer functionalities like navigation, positioning, mapping, and others"
to: "The stack is a collection of packages that offers functionalities like navigation, positioning, mapping, and much more"

File: docs/design_document/design.tex [hunk 4]
```diff
@@ -69,10 +69,14 @@ \section{Definitions}
...
 
 \subsubsection{Method}
-We will ROS will utilize as our method to manipulate the fetch robots sensors, positioning, mapping, and cameras. ROS libraries provide the system with the necessary tools to map an entire room so the fetch robot can familiarize itself with its test environment. Additionally, ROS will help us move the robot 360-degrees around the target object so that we are able to gather our training data set. ROS also provides us with methods to implement an image tagging technique, involving radio waves and RFID tags to locate a tagged object in its environment.
+We utilize ROS as our method to manipulate the fetch robots sensors, positioning, mapping, and cameras. ROS libraries provide the system with the necessary tools to map an entire room so the fetch robot can familiarize itself with its test environment. Additionally, ROS will help us move the robot 360-degrees around the target object so that we are able to gather our training data set. ROS also provides us with methods to implement an image tagging technique, involving radio waves and RFID tags to locate a tagged object in its environment.
```
Comment (julianweisbord) [inline docs/design_document/design.tex]: We utilize ROS as our method to manipulate the Fetch's sensors, positioning, mapping, and cameras.

File: docs/design_document/design.tex [hunk 5]
```diff
@@ -155,8 +160,8 @@ \subsubsection{Results and Data Analysis}
 This project uses DMA to ideally improve the agent's rate of classification over time. Without using DMA, the agent will only remember the information it learned in its last training epoch which most likely will not be as high of classification rate. This gives us something to compare DMA agianst.The results of using a method to mitigate catastrophic interference should be huge in comparison to an online learning technique that does not. We will conduct a quantitative 2-sample data analysis to show a significant increase in the rate of successful classification when using the Dual Memory Architecture.
```
Comment (julianweisbord) [inline docs/design_document/design.tex]: Let's change DMA to Elastic Weight Consolidation (EWC)

File: docs/design_document/design.tex [hunk 6]
```diff
@@ -155,8 +160,8 @@ \subsubsection{Results and Data Analysis}
...
...le to recognize and predict things about it's environment, this agent must first be trained. For this project, the agent is trained on images from varying angles of common objects in a lab environment. The agent will be provided with pictures from 5 object classes: mugs, staplers, chairs, books, and screwdrivers. Once several images are gathered and labeled, the agent will train itself on this dataset. This is the first step in actual image recognition \cite{ImgRecog}. The main overarching classifier is trained on images from the 5 object classes. This classifier will output a percentage likelihood for each object class and then will pass the highest class prediction percentage to sub-classifiers that can more accurately classify the object because they specialize in the specific features to that object class. Finally, there is an additional step to image recognition, that when implemented, allows the agent to continually learn new information about these objects over time, this is called Sequential Learning \cite{Kirkpatrick}. A DMA is implemented in order to maximize the amount of new and old information that the intelligent agent can encode, store, and retrieve \cite{McDermott}.
```
Comment (julianweisbord) [inline docs/design_document/design.tex]: Change DMA to EWC here too. Also maybe write a sentence about image noise around: "For this project, the agent is trained on images from varying angles of common objects in a lab environment."

**Review by julianweisbord — CHANGES_REQUESTED** [review]

**Comment by julianweisbord** [comment]
Any update to this?

**Comment by julianweisbord** [comment]
We need to get these changes approved by McGrath asap.

**Comment by Rodrigmi** [comment]
Have you done the changes that Julian suggested? Once those are done it should be good to go.

**Comment by mccallm1** [comment]
Document updated with suggestions in mind, should be good to approve and send out.

**Review by mccallm1 — COMMENTED** [review]
Good input: helps clarify wording and updated usage

**Review by mccallm1 — COMMENTED** [review]
Changes were added to file and PR is ready to be reviewed again

**Review by julianweisbord — APPROVED** [review]

**Pull request merged**